pkg/apis/capabilities/v1beta1/openapi_types.go

```diff
@@ -0,0 +1,193 @@
+package v1beta1
+
+import (
+	"reflect"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	OpenapiKind = "Openapi"
+
+	// OpenapiInvalidConditionType represents that the combination of configuration
+	// in the OpenapiSpec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	// Example: the spec references invalid openapi spec
+	OpenapiInvalidConditionType common.ConditionType = "Invalid"
+
+	// OpenapiReadyConditionType indicates the openapi resource has been successfully reconciled.
+	// Steady state
+	OpenapiReadyConditionType common.ConditionType = "Ready"
+
+	// OpenapiFailedConditionType indicates that an error occurred during reconcilliation.
+	// The operator will retry.
+	OpenapiFailedConditionType common.ConditionType = "Failed"
+)
+
+// OpenAPIRefSpec Reference to the OpenAPI Specification
+type OpenAPIRefSpec struct {
+	// ConfigMapRef ConfigMap that contains the OpenAPI Document
+	// +optional
+	ConfigMapRef *corev1.ObjectReference `json:"configMapRef,omitempty"`
+
+	// URL Remote URL from where to fetch the OpenAPI Document
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.*$`
+	// +optional
+	URL *string `json:"url,omitempty"`
+}
+
+// OpenapiSpec defines the desired state of Openapi
+type OpenapiSpec struct {
+	// OpenAPIRef Reference to the OpenAPI Specification
+	OpenAPIRef OpenAPIRefSpec `json:"openapiRef"`
+
+	// ProviderAccountRef references account provider credentials
+	// +optional
+	ProviderAccountRef *corev1.LocalObjectReference `json:"providerAccountRef,omitempty"`
```

Comment:
Are this other fields defined because are not supported natively in OpenAPI spec?

Reply:
Yes, these other fields are 3scale specific configuration beyond OpenAPI spec. Taken from 3scale toolbox. 

However, 3scale product full configuration is not yet covered. Application plans and other settings are still missing.

Reply:
:+1: